static/js/emoji_picker.js

```diff
@@ -703,6 +704,11 @@ export function register_click_handlers() {
     $("body").on("click", ".emoji_map", function (e) {
         e.preventDefault();
         e.stopPropagation();
+        const compose_control_buttons_popover = popover_menus.get_compose_control_buttons_popover();
+        if (compose_control_buttons_popover) {
+            e.target = compose_control_buttons_popover.reference;
+        }
```

Comment:
Monkey-patching `e.target` feels risky as a code pattern; I don't know what browser guarantees there are about whether other events that this might propagate to are affected.

It's probably OK as a temporary regression since these click handlers generally have a `stopPropagation` (`video_link` doesn't, which is probably its own bug).  And I think this PR is complex enough and enough of an improvement to the current `main` we should merge and fix forward.

Reply:
I think actually the best way to do this is to extract a shared function `extract_compose_click_target` that we can just have these 4-5 different code paths share, and takes care of the `popover_menus` logic and perhaps other details like returning an `edit_message_id`.  And then each of these individual pieces of code can just work with those elements?  Not sure, but in any case probably deserves its own dedicated refactoring PR.